editoast/src/valkey_utils.rs

```diff
@@ -273,6 +273,46 @@ impl ValkeyConnection {
         });
         Ok(cached_values)
     }
+
+    /// Add one serializable member to a sorted set, or update its score if it already exists.
+    pub async fn json_zadd<
+        K: Debug + ToRedisArgs + Send + Sync,
+        S: ToRedisArgs + Send + Sync,
+        M: Serialize,
+    >(
+        &mut self,
+        key: K,
+        member: M,
+        score: S,
+    ) -> Result<()> {
+        let str_member = match serde_json::to_string(&member) {
+            Ok(member) => member,
+            Err(_) => {
+                return Err(RedisError::from((
+                    ErrorKind::IoError,
+                    "An error occurred serializing to json",
+                ))
+                .into());
+            }
+        };
+        self.zadd::<_, _, _, ()>(key, str_member, score).await?;
+        Ok(())
+    }
+
+    /// Return a range of members in a sorted set, by score.
+    pub async fn json_zrangebyscore<
+        K: ToRedisArgs + Send + Sync,
+        M: ToRedisArgs + Send + Sync,
+        MM: ToRedisArgs + Send + Sync,
+    >(
+        &mut self,
+        key: K,
+        min: M,
+        max: MM,
+    ) -> Result<Vec<String>, RedisError> {
```

Comment:
This should return a list/iterator of `T: DeserializeOwned`, not just Strings. See `json_get_bulk()` ;)

Reply:
I'm curious, why was this thread resolved?

Reply:
Because `json_zrangebyscore()` no longer returns a `Result<Vec<String>, RedisError>`? ie. it's been fixed?

Or am I missing something?

Reply:
Yes it does return a `Vec`.

Reply:
Ok, it seems I missunderstood the main point of this comment, and so you're right, the point you raised has been fixed, I focused on the word `Iterator`. I'll open another comment.